be/src/exec/pipeline/pipeline_driver_poller.cpp

```diff
@@ -192,6 +192,19 @@ size_t PipelineDriverPoller::activate_parked_driver(const ImmutableDriverPredica
     return ready_drivers.size();
 }
 
+size_t PipelineDriverPoller::calculate_parked_driver(const ImmutableDriverPredicateFunc& predicate_func) const {
+    size_t parked_driver_num = 0;
+    auto driver_it = _parked_drivers.begin();
+    while (driver_it != _parked_drivers.end()) {
+        auto driver = *driver_it;
+        if (predicate_func(driver)) {
+            parked_driver_num += 1;
+            driver_it++;
```

Comment:
why ++ here? dead loop?

Reply:
I think it's used to iterate all parked drivers, in the list

Reply:
fixed.